extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -178,11 +221,53 @@ export class AmpAutocomplete extends AMP.BaseElement {
     this.action_ = Services.actionServiceForDoc(this.element);
     this.viewport_ = Services.viewportForDoc(this.element);
 
+    const inputElements = childElementsByTag(this.element, 'input');
+    const textareaElements = childElementsByTag(this.element, 'textarea');
+    userAssert(
+      inputElements.length + textareaElements.length === 1,
+      `${TAG} should contain exactly one <input> or <textarea> child`
+    );
+    this.inputElement_ =
+      /** @type {!HTMLInputElement} */ (inputElements[0] ||
+      textareaElements[0]);
+    if (this.inputElement_.hasAttribute('type')) {
+      const inputType = this.inputElement_.getAttribute('type');
+      userAssert(
+        inputType === 'text' || inputType === 'search',
+        `${TAG} requires the "type" attribute to be "text" or "search" if present on <input>`
+      );
+    }
+
+    this.inline_ = this.element.hasAttribute('inline');
```

Comment:
You don't need this to be an instance prop since you only use it locally, make it a `const`.